src/prog_models/prognostics_model.py

```diff
@@ -1296,3 +1298,75 @@ def generate_surrogate(self, load_functions, method = 'dmd', **kwargs):
             raise ProgModelInputException(f"Invalid 'event_keys' input value ({config['event_keys']}), must be a subset of the model's events ({self.events}).")
 
         return SURROAGATE_METHOD_LOOKUP[method](self, load_functions, **config)
+
+    class CustomEncoder(json.JSONEncoder):
+        """
+        Custom encoder to serialize parameters 
+        """
+        def default(self, o):
+            if isinstance(o, np.ndarray):
+                return {'original_type': 'ndarray', 'data': o.tolist()}
+            elif isinstance(o, DictLikeMatrixWrapper):
+                dict_temp = {o.keys()[iter]: o[o.keys()[iter]] for iter in range(len(o.keys()))}
+                dict_temp['original_type'] = 'DictLikeMatrixWrapper'
+                return dict_temp 
+            else: 
+                from base64 import b64encode
+                pkl_temp = b64encode(pickle.dumps(o))
+                save_temp = {}
+                save_temp['data'] = pkl_temp.decode()
+                save_temp['original_type'] = 'pickled'
+                return save_temp
+    
+    def to_json(self):
+        """
+        Serialize parameters to save as JSON objects 
+
+        Note
+        ----
+        This method only serializes the values in the parameters class of the prognostics model
+        """
+        parameters_dict = {}
+        for key in self.parameters.keys():
+            parameters_dict[key] = self.parameters[key]
+
+        return json.dumps(parameters_dict, cls=self.CustomEncoder)
+    
+    @classmethod
+    def from_json(cls,data):
+        """
+        Create a prognostics model model from a previously generated model that was serialized as a JSON object
+
+        Args:
+            data: 
+                JSON serialized parameters necessary to build a model 
+                See to_json method 
+
+        Returns:
+            PrognosticsModel: Model generated from serialized parameters 
+
+        Note
+        ----
+        This serialization only works for models that include all parameters necessary to generate the model in the parameters class. 
+        """
+        def custom_decoder(o):
+            """
+            Custom decoder to deserialize parameters 
+            """
+            if isinstance(o,dict) and 'original_type' in o.keys():
+                if o['original_type'] == 'ndarray':
+                    return np.array(o['data'])
+                elif o['original_type'] == 'DictLikeMatrixWrapper':
+                    del o['original_type']
+                    return DictLikeMatrixWrapper(list(o.keys()),o)
+                elif o['original_type'] == 'pickled':
+                    import pickle
+                    from base64 import b64decode
+                    pkl_temp1 = o['data'].encode()
+                    pkl_temp2 = b64decode(pkl_temp1)
+                    return pickle.loads(pkl_temp2)
+            return o
```

Comment:
Let's add an else statement here to the original type check that will raise an exception. Something like f"Type {o['original_type']} not supported by PrognosticsModel json decoder